frontend/src/page/profile.tsx

```diff
@@ -25,7 +27,39 @@ export const Profile: React.FC = () => {
     return <div>Loading</div>;
   }
   return (
-    <div>
-        Username: <input type="text" value={user.username}/>
-    </div>);
+    <form>
+      <TextField
+        fullWidth
+        label="Username"
+        value={user.username}
+        onChange={(e) => {
+          user.username = e.target.value;
+          setUser(user);
+        }}/>
+      <TextField
+        fullWidth
+        label="Bio"
+        value={user.bio}
+        onChange={(e) => {
+          user.bio = e.target.value;
+          setUser(user);
+        }}/>
+      <Button
+        type="submit"
+        onClick={async (e) => {
+          e.preventDefault();
+          const profileUpdateFetchPromise = abortableJsonFetch<StoredUser>(
```

Comment:
One thing I'd note here is that if the user goes away from the profile page to another route, then when this completes it will attempt to call `setUser` below, which results in a React warning about modifying state on an unmounted component (because the profile page is gone). This was the entire reason I created `abortableJsonFetch` actually. The solution here, since it's in an `onClick` handlers is to add another state that is the `profileUpdateFetchPromise` and call `cancel`. The `thread.tsx` does something similar when you post a comment, since you can technically click away before it finishes. So above you'd put:
```ts
  const [profileUpdateFetch, setProfileUpdateFetch] = React.useState<AbortablePromise<StoredUser>>(null);
```
Then right when you create the promise you'd put:
```ts
setProfileUpdateFetch(profileUpdateFetchPromise);
```

And then again up above... this snippet is ugly but it is the React functional component way of saying "run this code when the component is unmounted":
```ts
  React.useEffect(() => () => {
    cancel(profileUpdateFetch);
  }, []);
```

Reply:
I added the code [here](https://github.com/TrevorSundberg/madeitforfun/pull/9/commits/c895f04efe43f2190f06db573aa2a9631a8bc02a), I'm not totally sure how to test it well though to be honest. The save happens so fast its difficult to try to navigate away while it is happening.

Reply:
Yeah that's alright, it looks correct. I probably should build in a way to delay fetches for testing.